runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -819,27 +849,32 @@ ClientSessionData::getClassRecord(ClientSessionData::ClassInfo &classInfo)
 }
 
 const AOTCacheClassRecord *
-ClientSessionData::getClassRecord(J9Class *clazz, bool &missingLoaderRecord)
+ClientSessionData::getClassRecord(J9Class *clazz, bool &missingLoaderRecord, bool &uncachedClass)
    {
    TR_ASSERT(getROMMapMonitor()->owned_by_self(), "Must hold ROMMapMonitor");
 
    auto it = getROMClassMap().find(clazz);
    if (it == getROMClassMap().end())
+      {
+      uncachedClass = true;
       return NULL;
+      }
 
-   auto record = getClassRecord(it->second);
+   bool missingLoaderInfo = false;
```

Comment:
Should simply pass down the reference to the flag passed as argument (which should be renamed), otherwise we lose this error information here.